### ISSUE TITLE
Multiple change listeners result in wrong column names

### ISSUE THREAD

**Issue by roberthecker** [cbl-dart/cbl-dart#322]
**Describe the bug**

Firstly, I am unsure whether this is an actual bug or just a misunderstanding on my part on how the API works/what results are to be expected when using the `changes()` method on a `Query`.

Our project uses multiple queries to listen to different data types which are differentiated through a `type` field on each document. We create a query filtering by that field and listen for changes to the results using `changes()`.  As long as we only listen to a single query, the change listener receives valid documents in its callback and we can parse those results into our internal data structure. 
As soon as multiple queries are being listened to, the change listener of one query receives `Result`s which contain the correct document data of the correct `type` but with incorrect column names. The log messages show that the correct column names are being received on the native side but end up incorrect in the `ResultSet`.

**To Reproduce**

1. Create at least 2 queries filtering by some kind of type field such as 
```dart
Query modelAQuery = const QueryBuilder()
        .select(SelectResult.all(), SelectResult.expression(Meta.id))
        .from(DataSource.database(_db))
        .where(Expression.property('type').equalTo(Expression.string(ModelA.typeIdentifier)));
Query modelBQuery = const QueryBuilder()
        .select(SelectResult.all(), SelectResult.expression(Meta.id))
        .from(DataSource.database(_db))
        .where(Expression.property('type').equalTo(Expression.string(ModelB.typeIdentifier)));
```
2. Listen to changes on both queries (only one shown here for brevity, the content of the callback is irrelevant but shows where we run into parsing problems on our side)
```dart
_modelASub = modelAQuery
        .changes()
        .map((change) => change.results.asStream().map((result) {
              Map<String, Object?> plainMap = result.toPlainMap();
              ModelA aInstance = ModelA.fromJson(plainMap[_dbName] as Map<String, Object?>);
              aInstance.id = plainMap['id'] as String;
              return aInstance;
            }).toList())
        .listen((aInstancesFuture) async {
      List<ModelA> aInstances = await aInstancesFuture;

      if (mounted) {
        setState(() {
          _aInstances = aInstances;
        });
      }
    });
```
3. Insert some data with one type and then the other type to trigger the change listeners.
4. Use a logger with `debug` level 
```dart
    Database.log
      ..console.level = LogLevel.debug
      ..custom = DartConsoleLogger(LogLevel.debug);
```
5. Use a breakpoint after `result.toPlainMap()` has been called and compare the fields of `plainMap` to the output of what the native side logged.

**Expected behavior**

The keys of the map should match the keys of the document. Instead, they contain values which are valid keys for a document with a different `type` but not this one.

**Screenshots**

***Adding a document with type A (works as expected)***
This is the console output on a small demo application when adding only documents of one type. This works as expected.
![image](https://user-images.githubusercontent.com/7288631/157280757-4b4c167a-8b30-48e5-a2a8-5108301d2dd1.png)

The debugger shows matching values for this document inside of the change listener:
![image](https://user-images.githubusercontent.com/7288631/157280642-99ae24d4-459c-4c1e-86d9-ecbd1b76c4b1.png)

***Adding a document with type B afterwards***
Adding a document with a different type results in this console output:
![image](https://user-images.githubusercontent.com/7288631/157281638-2b0167fa-c440-4d6c-8821-4597de33547d.png)
The column names of this output are correct and match our defined data structure, we expect the same column names in the result object passed to the change listener.

Instead, the debugger shows that the result contains different column names (both in some internal fields and in the result of `toPlainMap()`):
![image](https://user-images.githubusercontent.com/7288631/157282575-9049fdd3-be2d-40e0-b132-2f595631c0cc.png)
Interestingly, the values of `columnValuesArray` have the correct column names while the ones in `_array` are wrong.

**Environment (please complete the following information):**

 - OS: Android
 - `cbl`: 1.0.0+1
 - `cbl_flutter`: 1.0.0+1
 - `cbl_flutter_ce`: 1.0.0

**Additional context**

A small repository with a demo app illustrating this problem can be found [here](https://github.com/roberthecker/cbl_flutter_confusion)

**Comment by blaugold** [comment]
Thanks for the detailed report! This is definitely a bug.

**Comment by blaugold** [comment]
I've published a hotfix release for `cbl` (`v1.0.1`) with a fix.

**Comment by nblum37** [comment]
Hi @blaugold,

we have a probably similar problem with cbl 1.0.3. Sometimes the value (List type) of a specific key is copied to another key in every document. However, this problem occurs only at around 50% of all initial database syncs. It is very hard to determine external factors, that influence the wrong behavior.

This is the usual document structure for all database items
![Bildschirmfoto 2022-03-17 um 02 23 36](https://user-images.githubusercontent.com/82163574/158718962-c52bb0c5-a68b-4358-a205-25da76545461.png)

However, at 50% of all initial synchronization, the value of the map entry 9 is copied to entry 0 (the behavior applied to all 60 replicated documents).
![Bildschirmfoto 2022-03-17 um 02 29 50](https://user-images.githubusercontent.com/82163574/158719237-cb1624b8-8c9a-41e8-9048-0e5ab54ecf86.png)

The documents are directly loaded from the database after being replicated. The `await db.document(documentId)` function is called by the only existing change listener of the database. However, we have a second cbl instance with a separate change listener running in parallel but without a replicator. The only thing we observe (but not sure if it's relevant) is that usually the problem only occurs if both databases are initialized at app startup. If we only delete and re-sync the replicated database and leave the other database untouched, then the mentioned behavior is very unlikely. I don't see any strange behavior in the database without replicator.
I hope, the error description can give you some idea of the problem. If you need any further information, pls let me know.

**Comment by blaugold** [comment]
Hey @nblum37. I have created a new issue (#329) since this seems like a different issue to me.
